### ISSUE TITLE
Intersection not 3D Compatible

### ISSUE THREAD

**Issue by dalexander-trc** [bjornharrtell/jsts#339]
## Description of the Issue
When intersecting two geometries, I would expect the z values to be interpolated in the same way that the x and y values are. They currently are set to undefined. Additionally, it seems like Coordinate's are set up to support z values, but the parsers cannot interpret them.

### Intent
We plan on adding another attribute, m, to the Coordinate class and will want that interpolated as well. It seems like before we can do that we need to determine the extent that z is supported and add to that if necessary. I've also summarized this in the [Google Group](https://groups.google.com/forum/#!topic/jsts-devs/3xsHJhV9vLk), but there hasn't been very much activity there.

**my-test.js**
```
const jsts = require("jsts");
const WKTReader = new jsts.io.WKTReader();

var simpleString = "LINESTRING Z (0 5 3,10 5 5)";
var simplePolygon = "POLYGON Z ((6 1 3,8 1 3,8 7 3,6 7 3,6 1 3))";

// Pretending that the WKTReader can interpret Z values
var jString = WKTReader.read(simpleString);
var jPolygon = WKTReader.read(simplePolygon);

var intersection = jString.intersection(jPolygon);
```

### Actual Result
A geometry containing two Coordinates is returned. One at (6,5) and another at (8,5). Both have "undefined" for their z value. 

### Expected Result
A geometry containing two Coordinates is returned. One at (6,5) and another at (8,5). Because the input geometries contain z values, the output geometry will contain interpolated z values, something like 3.6 and 3.8, respectively.

**Comment by bjornharrtell** [comment]
This is out of scope of the project. See answer on the google group discussion for further information.
